### PR TITLE
Merge and update developer notes and published help page about search ranking

### PR DESCRIPTION
**Pull request by isoos**
- #5035
- Mostly merge, and slight update to more recent terms and weights.

### CODE REVIEW

**Comment by kwalrath** [comment]
I plan to take a look at this tomorrow.

**Comment by kwalrath** [comment]
@mit-mit was planning to take a look, so I'll wait for him to finish before I do my pass.

File: doc/help-search.md [hunk 1]
```diff
@@ -24,17 +24,150 @@ The search UI also supports filters:
...
+matches the platforms of the package, and also calculate how closely it
+matches them ([`scope_specificity.dart`](https://github.com/dart-lang/pub-dev/blob/master/app/lib/search/scope_specificity.dart)).
+
+For example, if the platform filter is `platform:web`, we'll have the following scores:
+
+- `<missing or empty platform>`: no match, doesn't show up in the results
+- `['platform:android', 'platform:ios']`: no match, doesn't show up in the results
+- `['platform:web']`: `1.0` (specific platform match)
+- `['platform:android', 'platform:web']`: `0.9` (1 extra platform, close platform match)
+- `['platform:android', 'platform:ios', 'platform:web']`: `0.8` (2+ extra platforms, distant platform match)
```
[the remark this answers is not in the document]
Comment (isoos) [reply doc/help-search.md:61]: > Are you telling me we rank packages lower if they support additional platforms?

Yeah, in some cases, we do. E.g. if a query filters on X, and pkg1 supports X, pkg2 supports X, Y, and Z, pkg3 supports X and Z, then their relative order (all other being equal) is: pk1, pkg3, pkg2.
https://github.com/dart-lang/pub-dev/blob/master/app/test/search/scope_specificity_test.dart

This is coming from a very early tuning of listing Flutter packages. As Any/Dart packages are also listed there, we wanted to slightly prefer Flutter-only ones under `/flutter/packages`.

File: doc/help-search.md [hunk 2]
```diff
@@ -24,17 +24,150 @@ The search UI also supports filters:
...
+
+### Text match ranking
+
+The indexing process prepares the package documents and parses the following fields:
+
+- package name
+- package description (first 500 characters)
+- readme (first 5000 characters)
+
+When a text query is specified, we'll try to match the query against all of these
```
Comment (mit-mit) [inline doc/help-search.md:79]: the word "try" is a bit odd here (hopefully the search does a bit more than trying?)

**Comment by isoos** [comment]
Closing as #5129 updated the dev documentation, and public help file will get updated separately.

**Pull request closed without merging**